test/test_line.py

```diff
@@ -285,6 +285,92 @@ def test_attrib_b(self):
         with self.assertRaises(AttributeError):
             del line.b
 
+    def test_attrib_midpoint(self):
+        """a full test for the midpoint attribute"""
+        expected_x1 = 10.0
+        expected_y1 = 2.0
+        expected_x2 = 5.0
+        expected_y2 = 6.0
+        expected_a = expected_x1, expected_y1
+        expected_b = expected_x2, expected_y2
+        expected_midpoint = (expected_x1 + expected_x2) / 2, (
+            expected_y1 + expected_y2
+        ) / 2
+        line = Line(expected_a, expected_b)
+
+        self.assertEqual(line.midpoint, expected_midpoint)
+
+        line.midpoint = expected_midpoint[0] - 1, expected_midpoint[1] + 1.321
+
+        self.assertEqual(
+            line.midpoint, (expected_midpoint[0] - 1, expected_midpoint[1] + 1.321)
+        )
+
+        line = Line(0, 0, 1, 0)
+
+        for value in (None, [], "1", (1,), [1, 2, 3], 1, 1.2):
+            with self.assertRaises(TypeError):
+                line.midpoint = value
+
+        with self.assertRaises(AttributeError):
+            del line.midpoint
+
+    def test_attrib_midpoint_x(self):
+        """a full test for the midpoint_x attribute"""
+        expected_x1 = 10.0
+        expected_y1 = 2.0
+        expected_x2 = 5.0
+        expected_y2 = 6.0
+        expected_a = expected_x1, expected_y1
+        expected_b = expected_x2, expected_y2
+        expected_midpoint = (expected_x1 + expected_x2) / 2, (
+            expected_y1 + expected_y2
+        ) / 2
+        line = Line(expected_a, expected_b)
+
+        self.assertEqual(line.midpoint_x, expected_midpoint[0])
+
+        line.midpoint_x = expected_midpoint[0] - 1
+
+        self.assertEqual(line.midpoint_x, expected_midpoint[0] - 1)
+
+        line = Line(0, 0, 1, 0)
+
+        for value in (None, [], "1", (1,), [1, 2, 3]):
```

Comment:
We should create a standardized variable for checking these integers & floats in the future, this seems to be a common copy-paste scenario.

Reply:
YEah i agree, some tests are a little more specific tho, so we'll most likely need more than one variant